src/CoreWCF.ConfigurationManager/src/WrappedConfigurationFile.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+
+namespace CoreWCF.Configuration
+{
+    internal class WrappedConfigurationFile
+    {
+        private static readonly (string name, Type type)[] s_knownSections = new[]
+{
+            ("services", typeof(ServicesSection)),
+            ("bindings", typeof(BindingsSection)),          
+        };
+
+        private readonly Lazy<string> _config;
+        public string ConfigPath => _config.Value;
+
+        public WrappedConfigurationFile(string path)
+        {
+            _config = new Lazy<string>(() => WrapFile(XDocument.Load(path)), true);
+        }
+
+        private static string WrapFile(XDocument original)
+        {
+            string configPath = Path.GetTempFileName();
+            // for debug
```

Comment:
We can't save this to a temp file. Often the temporary file path is readable by any process on the machine. The passed in config could have security keys included inline. I see this eventually used with ConfigurationFileMap which is meant to represent machine.config. This isn't the optimal way to do this. On .NET Framework, you have machine.config, and you have app.config (or web.config). When you use ConfigurationManager to look for things, it will effectively merge the two in the in-memory model. The correct thing to do would be to have the machine.config pieces in one file, and leave the input config file as is unaltered and in it's original location. That way the security of the app config is the responsibility of the service owner.  
There are a couple of options. You could ship a static machine.config file as part of the package and reference it. Or you could add a static machine config file as an assembly file resource and access that. Or you could dynamically create one in a temporary folder. I'm not a fan of the temporary file creation as you aren't guaranteed it will be deleted either.

Reply:
Yes, I understand that a hack with a temporary file is a bad idea, but this is just to force the ConfigurationManager to read a config like <system.serviceModel>,
if there is a way to do it InMemory (without creating your own custom sections) - let me know, I saw you provided the code below, I have not tested it yet.

Reply:
This turned out to be more complicated than I expected, there were multiple hurdles to overcome to get it to work. If you look at my [commit here](https://github.com/mconnew/CoreWCF/commit/080a51a33d0ebe83ee3b3faca75277bbf6481c3d) I have added a machine.config file to the nuget package and added a targets file to copy it into the application output folder. Then in the code I specify that as the machine.config file.

Reply:
@mconnew Great! thank you so much, I applied these changes in PR, it works, though it took a small fix for .net core 2.1 on linux